### ISSUE TITLE
Graphql -  context.unwrap(DataFetchingEnvironment.class) is always null

### ISSUE THREAD

**Issue by magick93** [quarkusio/quarkus#24121]
### Describe the bug

I'm trying to use `context` as described on https://quarkus.io/guides/smallrye-graphql:

> This context object also allows you to fall down to the underlying [graphql-java](https://www.graphql-java.com/) features by using the leaky abstraction:

### How to Reproduce?

Reproducer

1. Add `quarkus.smallrye-graphql.events.enabled=true` to properties file
2. Ensure smallrey-graphql extension is added. 
3. Add the following to a class with the `@GraphQLApi` annotation:

```
    DataFetchingEnvironment dfe = context.unwrap(DataFetchingEnvironment.class);
        /***
         * 
        java.lang.RuntimeException: java.lang.NullPointerException: 
        Cannot invoke "io.smallrye.graphql.execution.context.SmallRyeContext.unwrap(java.lang.Class)" 
        because the return value of "io.smallrye.graphql.execution.context.SmallRyeContext.getContext()" is null

         */
    
    public GraphQLSchema.Builder addMyOwnEnum(@Observes GraphQLSchema.Builder builder) {

        

        // Here add your own features directly, example adding an Enum
        GraphQLEnumType myOwnEnum = GraphQLEnumType.newEnum()
                .name("SomeEnum")
                .description("Adding some enum type")
                .value("value1")
                .value("value2").build();
    
        return builder.additionalType(myOwnEnum);
    }
```

### Output of `uname -a` or `ver`

Linux 5.13.0-30-generic #33-Ubuntu SMP Fri Feb 4 17:03:31 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.1" 2021-10-19 OpenJDK Runtime Environment Temurin-17.0.1+12 (build 17.0.1+12) OpenJDK 64-Bit Server VM Temurin-17.0.1+12 (build 17.0.1+12, mixed mode, sharing)

### Quarkus version or git rev

2.7.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)

**Comment by phillip-kruger** [comment]
@magick93 thanks, can you share a reproducer? Thanks

**Comment by magick93** [comment]
Hi @phillip-kruger - please see https://github.com/magick93/quarkus-smallrye-graphql-24121/tree/master

**Comment by phillip-kruger** [comment]
You have to use context inside a method. You are trying to use it as a class var
 The context is for a certain request, so use it somewhere in the flow of that request

**Comment by magick93** [comment]
Yes, I tried using in in both - in a method - in the `addMyOwnEnum` and as
a class var.

In both cases I had the same result.

**Comment by phillip-kruger** [comment]
The addmyown method is not part of the request flow. The context hold information about the request. So any method that is a query or mutation, or any method called by those further downstream, you can use context.

Addmyown is called on schema creation on startup, only once, due to the obeserver

**Comment by magick93** [comment]
Thanks Philip

The `addmyown` does not work as expected, too. There is no addition of an
enum to the graphql schema, which is what I thought this method would do.

**Comment by phillip-kruger** [comment]
Ok I'll have a look, that feature is under test so should be working. I'll get back to you

**Comment by phillip-kruger** [comment]
Hi @magick93 - Everything seems to be working fine. Once you move the unwrap to a method in the request flow, the app starts up and the "addmyown" also works. See: 
![working](https://user-images.githubusercontent.com/6836179/157149808-6f8ff887-7509-453e-9ee8-b615958f788d.gif)
